### PR TITLE
kernel: GPIO changes

### PR DESCRIPTION
**Pull request by stintel**
PR for CI. If interested in review, read individual commit messages. Will send this to the openwrt-devel ML for review if CI is green.

### CODE REVIEW

**Comment by Ansuel** [comment]
@stintel seems some config are still missing for some target

**Comment by stintel** [comment]
@Ansuel I've noticed. More targets use GPIO without actually having gpio in FEATURES. Will need to rethink this a bit.

File: target/linux/armvirt/Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ include $(TOPDIR)/rules.mk
 
 BOARD:=armvirt
 BOARDNAME:=QEMU ARM Virtual Machine
-FEATURES:=fpu pci rtc usb
+FEATURES:=fpu gpio pci rtc usb
```
Comment (hauke) [inline target/linux/armvirt/Makefile:9]: The gpio feature will automatically be set if the kernel configuration contains GPIO_SUPPORT, see: https://github.com/openwrt/openwrt/blob/master/scripts/target-metadata.pl#L23

**Review by chunkeey — APPROVED** [review]
all true. The libgpiod v2 has been in development for more than two years now. I though that they would release a official libgpiod v2.0+ this year... but apparently not.

**Pull request closed without merging**